### PR TITLE
Add total row in report tables

### PR DESCRIPTION
**Pull request by K-Markopoulos**
Issue #178 

Add total row at the top of the table with the sum of the values.

- Calculate the aggregated values and ad extra row to the data
- Adjust sorting in order to always stay on top

### CODE REVIEW

File: apps/frontend/utils/tableFormatting.tsx [hunk 1]
```diff
@@ -85,7 +89,7 @@ const getLocationColumnSetup = (
         <FormattedMessage id={`forms_table.headers.${params.field}`} />
       </Typography>
     ),
-    valueFormatter: value =>
+    valueGetter: value =>
```
Comment (ericboucher) [inline apps/frontend/utils/tableFormatting.tsx:92]: Can you explicit why this change is needed?

**Comment by ericboucher** [comment]
@K-Markopoulos let's use the translation files so that we can get a translated version in Khmer: `សរុប`

File: apps/frontend/utils/tableFormatting.tsx (hunk 1 above)
Comment (K-Markopoulos) [reply apps/frontend/utils/tableFormatting.tsx:92]: The return value of `valueGetter` is used for sorting and it's being passed in the `valueFormatter` or `renderCell`. With this change:
1. We can use `params.value` in the `renderCell` without the need to duplicate the logic to construct the value.
2. The sorting is based on the displayed value (the actual name of the location) and not on the code ("01-002") and behaves as expected (unless we need a custom order based on the code and not alphanumerical order).

From [docs](https://mui.com/x/react-data-grid/column-definition/#value-formatter):
> Note, that the value returned by valueFormatter is only used for rendering purposes. Filtering and sorting are based on the raw value (row[field]) or the value returned by [valueGetter](https://mui.com/x/react-data-grid/column-definition/#value-getter).

Comment (K-Markopoulos) [reply apps/frontend/utils/tableFormatting.tsx:92]: We can use the same approach here to calculate `formattedList` once in the `valueGetter`

https://github.com/WFP-VAM/dmp/blob/74ec186e365f7ffe50ac225422c544ea5df6296b/apps/frontend/utils/tableFormatting.tsx#L121-L139

I skipped this as I thought it was out of scope and required more testing, but we can make that change since we are working on this area

Comment (ericboucher) [reply apps/frontend/utils/tableFormatting.tsx:92]: Ok, worth exploring. Two things come to mind though:
1. Ideally, for sorting, we might actually want to order by number of villages instead of the alphabetical order of the first village (which doesn't really mean anything)
2. We still need to display the tooltip

**Review by ericboucher — APPROVED** [review]

**Comment by wadhwamatic** [comment]
Thanks for this @K-Markopoulos and @ericboucher. Is there a reason there's a discrepancy in the total when you're showing Province vs Commune level views? I expected the values to be the same in the total row, but they are not. 

Province level:
Screenshot 2024-10-21 at 14 35 42

Commune level:
Screenshot 2024-10-21 at 14 35 49

**Comment by K-Markopoulos** [comment]
@wadhwamatic thanks for the catch! Indeed in commune level some rows are already aggregated and our logic would not take that into account. Hopefully the last commit will address this issue, the total numbers are now the same between these views.

File: apps/frontend/utils/tableFormatting.tsx [hunk 2]
```diff
@@ -359,31 +367,35 @@ export const useAggregatedRow = <
...
-      return acc;
-    },
-    { id: TOTAL_ROW_ID },
-  );
+  const aggregatedRow = data
+    .filter(rowFilter)
+    .reduce<Record<string, string | number | string[]>>(
+      (acc, row) => {
+        columns.forEach(({ type, field }) => {
+          if (field === KoboCommonKeys.village) {
```
Comment (ericboucher) [inline apps/frontend/utils/tableFormatting.tsx:384]: How about transforming to a set here directly?

File: apps/frontend/components/DisasterTable/CommuneLevelReportTable.tsx [hunk 3]
```diff
@@ -148,6 +148,9 @@ export const CommuneLevelReportTable = ({
       columnHeaderHeight="large"
       variant={disasterTableParams.variant}
       isFirstTable={disasterTableParams.isFirstTable}
+      aggregateRowFilter={row =>
```
Comment (ericboucher) [inline apps/frontend/components/DisasterTable/CommuneLevelReportTable.tsx:151]: Let's add a comment explaining why we need to do this

**Comment by wadhwamatic** [comment]
Looks good now, thanks @K-Markopoulos and @ericboucher!
